src/server/serve.rs

```diff
@@ -44,6 +53,10 @@ where
 
     loop {
         tokio::select! {
+            // A connection being accepted is more likely to occur than the
+            // server shutting down.
+            biased;
```

Comment:
I was originally thinking that it made sense to poll the future returned from the accept function (and also the connection) more frequently than `shutdown_rx.changed()`. However, I think the server benefits from having another future to poll. It creates natural backpressure and it's lock-free. I'm going to remove this.